marconi-chain-index/src/Marconi/ChainIndex/Indexers/MintBurn.hs

```diff
@@ -452,9 +453,15 @@ instance RI.Queryable MintBurnHandle where
           quantity
           redeemer
     where
+      -- filter the rows of a given predicate from in-memory events
+      filteredInMemoryRows :: (TxMintRow -> Bool) -> [TxMintRow]
+      filteredInMemoryRows f = filter f $ toRows =<< (coerce $ toList memoryEvents)
+      -- filter in-memory events for a given predicate
       filteredMemoryEvents :: [TxMintEvent]
-      filteredMemoryEvents = coerce $ fromRows $ filter rowFilter $ toRows =<< (coerce $ toList memoryEvents)
-
+      filteredMemoryEvents = coerce $ fromRows $ filteredInMemoryRows rowFilter
+      -- find txId's corresponding slotNo from in-memory events
+      findInMemroySlotNoFromTxId :: C.TxId -> Maybe C.SlotNo
+      findInMemroySlotNoFromTxId txid = fmap _txMintRowSlotNo . listToMaybe . filteredInMemoryRows $ (\r -> r ^. txMintRowTxId == txid)
```

Comment:
Minor: instead of `filter` (used in `filteredInMemoryRows`), `find` would be cleaner here (even if lazyness would probably lead to the same complexity)